libpod/image/pull_test.go

```diff
@@ -169,18 +177,56 @@ func TestPullGoalFromImageReference(t *testing.T) {
 				{"example.com/empty:latest", "example.com/empty:latest"},
 			},
 			true,
+			false,
+			false,
+		},
+		{ // Reference image by name in multi-image archive
+			"docker-archive:testdata/docker-two-images.tar.xz:example.com/empty:latest",
+			[]expected{
+				{"example.com/empty:latest", "example.com/empty:latest"},
+			},
+			true,
+			false,
+			false,
+		},
+		{ // Reference image by name in multi-image archive
+			"docker-archive:testdata/docker-two-images.tar.xz:example.com/empty/but:different",
+			[]expected{
+				{"example.com/empty/but:different", "example.com/empty/but:different"},
+			},
+			true,
+			false,
+			false,
+		},
+		{ // Reference image by index in multi-image archive
+			"docker-archive:testdata/docker-two-images.tar.xz:@0",
+			[]expected{
+				{"example.com/empty:latest", "example.com/empty:latest"},
+			},
+			true,
+			false,
+			false,
+		},
+		{ // Reference image by index in multi-image archive
+			"docker-archive:testdata/docker-two-images.tar.xz:@1",
+			[]expected{
+				{"example.com/empty/but:different", "example.com/empty/but:different"},
+			},
+			true,
+			false,
+			false,
 		},
-		{ // FIXME: Two images in a single archive - only the "first" one (whichever it is) is returned
-			// (and docker-archive: then refuses to read anything when the manifest has more than 1 item)
+		{ // Reference entire multi-image archive must fail (more than one manifest)
 			"docker-archive:testdata/docker-two-images.tar.xz",
-			[]expected{{"example.com/empty:latest", "example.com/empty:latest"}},
-			// "example.com/empty/but:different" exists but is ignored
+			[]expected{},
+			true,
+			false,
 			true,
 		},
 
 		// == oci-archive:
-		{"oci-archive:/dev/this-does-not-exist", nil, false}, // Input does not exist.
-		{"oci-archive:/dev/null", nil, false},                // Input exists but does not contain a manifest.
+		{"oci-archive:/dev/this-does-not-exist", nil, false, false, false}, // Input does not exist.
```

Comment:
I know in the error cases the `compareSourceRefs` value does not matter, but will later be hard to tell whether the `false` is an intentional property of `oci-archive` or not. “Everything but docker-archive has this set to `true`” is a simpler story to tell.

Reply:
Done and dropped a comment.